jfr-connection/src/main/java/io/opentelemetry/contrib/jfr/connection/FlightRecorderDiagnosticCommandConnection.java

```diff
@@ -210,26 +214,34 @@ public void closeRecording(long id) {
     throw new UnsupportedOperationException("closeRecording not supported on Java 8");
   }
 
+  // Do this check separate from assertCommercialFeatures because reliance
+  // on System properties makes it difficult to test.
+  private static boolean jdkHasUnlockCommercialFeatures() {
+    String javaVmVendor = System.getProperty("java.vm.vendor");
+    String javaVersion = System.getProperty("java.version");
+    return javaVmVendor.contains("Oracle Corporation") && javaVersion.matches("(?:^1\\.8|9|10).*");
```

Comment:
This got me to thinking... I think this code is wrong altogether since it tests the properties of the JVM running the agent and not that of the connected MBeanServerConnection. Now, that MBeanServerConnection is very likely to be the local JVM, but it could be a remote JVM as well.

Reply:
Pushed a change for using the MBeanServerConnection to get the vm version and vendor.